### PR TITLE
add net.danigm.timetrack

### PR DESCRIPTION
**Pull request by danigm**
This is a small app to do timetrack for GNOME. This app tracks the working time and stores in a sqlite database in your system so you can view daily, weekly or monthly reports.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.gnome.timetrack

**Comment by nedrichards** [comment]
I'm not here to enforce the guidelines, but is this app "official GNOME software", otherwise it probably shouldn't have the org.gnome app-id. I only say this because it's really annoying for flatpak users if apps change their app-id after being published.

File: org.gnome.timetrack.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        "--talk-name=ca.desrt.dconf",
+        "--talk-name=org.gtk.vfs",
+        "--talk-name=org.gtk.vfs.*",
+        "--talk-name=org.freedesktop.Notifications"
+    ],
+    "modules" : [
+        {
+            "name" : "timetrack",
+            "buildsystem" : "meson",
+            "builddir" : true,
```
Comment (bilelmoussaoui) [inline org.gnome.timetrack.json]:
```suggestion
```

File: org.gnome.timetrack.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--filesystem=xdg-run/gvfs",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+        "--talk-name=ca.desrt.dconf",
+        "--talk-name=org.gtk.vfs",
```
Comment (bilelmoussaoui) [inline org.gnome.timetrack.json]: are the gtk.vfs really needed?

Comment (danigm) [reply org.gnome.timetrack.json]: I don't think so, the app only need access to the filesystem to write the database file here:

```
os.path.join(GLib.get_user_data_dir(), "timetrack.sqlite3")
```

**Comment by nedrichards** [comment]
bot, build org.gnome.timetrack
testbot, build org.gnome.timetrack

**Comment by nedrichards** [comment]
@danigm cool, I've just checked with @csoriano1618 on IRC since that wasn't quite what I'd heard before but it seems like the projects view has evolved a little bit.
```csoriano> it's fine to use the gnome in id or technical terms, it's not fine to use GNOME as part of the end user branding. So the app ideally shouldn't be named GNOME something```
So that all sounds good to me, sorry for the interruption and we'll get on with getting this app into flathub!

**Comment by nedrichards** [comment]
bot, build org.gnome.timetrack
testbot, build org.gnome.timetrack

**Comment by danigm** [comment]
I've finally decided to use a personal app-id (net.danigm.timetrack) instead of the gnome one, because it seems that it's not recommended to use by default for now.

So @nedrichards, sorry for the noise. In any case, the discussion was useful for me and I think that will be useful for future apps with the same issue.

**Comment by bilelmoussaoui** [comment]
Ops, used the old id.
testbot, net.danigm.timetrack

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/net.danigm.timetrack

**Pull request closed without merging**